### PR TITLE
OMS Adapter: add new adapter

### PR DESCRIPTION
**Pull request by prBigBrother**
## Type of change

- [x] New bidder adapter   

## Description of change
In line with our rebranding efforts, I've submitted a pull request introducing a new adapter, OMS, mirroring the functionality of BrightcomSSP. This addition aims to ensure a seamless transition for our partners.

Simultaneously, I've initiated a pull request for the removal of two previous adapters, Brightcom and BrightcomSSP, in the upcoming major release #10925. 

- contact email: alexandruc@onlinemediasolutions.com
- test parameters for validating bids:
```
{
      bidder: 'oms',
      params: {
        publisherId: 2141020
      }
}
```

### CODE REVIEW

File: modules/omsBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+      deepSetValue(payload, 'regs.ext.gdpr', +bidderRequest.gdprConsent.gdprApplies);
+      deepSetValue(payload, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    }
+
+    if (bidderRequest?.uspConsent) {
+      deepSetValue(payload, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+    }
+
+    if (config.getConfig('coppa') === true) {
+      deepSetValue(payload, 'regs.coppa', 1);
```
Comment (patmmccann) [inline modules/omsBidAdapter.js:99]: Please get this from the request

File: modules/omsBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+          requestId: bid.impid,
+          cpm: parseFloat(bid.price),
+          width: parseInt(bid.w),
+          height: parseInt(bid.h),
+          creativeId: bid.crid || bid.id,
+          currency: 'USD',
+          netRevenue: true,
+          mediaType: BANNER,
+          ad: _getAdMarkup(bid),
+          ttl: 60,
```
Comment (patmmccann) [inline modules/omsBidAdapter.js:154]: Why not get this from the response

File: modules/omsBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+    }
+  }
+
+  bbox.width = bbox.right - bbox.left;
+  bbox.height = bbox.bottom - bbox.top;
+
+  return bbox;
+}
+
+function _getPercentInView(element, topWin, {w, h} = {}) {
```
Comment (patmmccann) [inline modules/omsBidAdapter.js:284]: This seems generally useful, please move it to a library and import it

File: modules/omsBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+}
+
+function _trackEvent(endpoint, data) {
+  ajax(`${TRACK_EVENT_URL}/${endpoint}`, null, JSON.stringify(data), {
+    method: 'POST',
+    withCredentials: false
+  });
+}
+
+function _isMobile() {
```
Comment (patmmccann) [inline modules/omsBidAdapter.js:204]: User agents are going away, please also inspect the device.sua object on the request

File: modules/omsBidAdapter.js [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+function getUserSyncs(syncOptions, responses, gdprConsent) {
+  return [];
+}
+
+function onTimeout(timeoutData) {
+  if (timeoutData === null) {
+    return;
+  }
+
+  _trackEvent('timeout', timeoutData);
```
Comment (patmmccann) [inline modules/omsBidAdapter.js:178]: Please put this behind configuration, publishers don't always want to provide

File: modules/omsBidAdapter.js [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+
+      return imp;
+    })
+
+    const referrer = bidderRequest?.refererInfo?.page || '';
+    const publisherId = getBidIdParameter('publisherId', bidReqs[0].params);
+
+    const payload = {
+      id: getUniqueIdentifierStr(),
+      imp: impressions,
```
Comment (patmmccann) [inline modules/omsBidAdapter.js:73]: Why not get the tid and gpid

File: modules/omsBidAdapter.js [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+            viewability: viewabilityAmountRounded
+          }
+        },
+        tagid: String(bid.adUnitCode)
+      };
+
+      const bidFloor = _getBidFloor(bid);
+
+      if (bidFloor) {
+        imp.bidfloor = bidFloor;
```
Comment (patmmccann) [inline modules/omsBidAdapter.js:62]: Might want the currency too

File: modules/omsBidAdapter.js [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+      },
+      tmax: bidderRequest?.timeout
+    };
+
+    if (bidderRequest?.gdprConsent) {
+      deepSetValue(payload, 'regs.ext.gdpr', +bidderRequest.gdprConsent.gdprApplies);
+      deepSetValue(payload, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    }
+
+    if (bidderRequest?.uspConsent) {
```
Comment (patmmccann) [inline modules/omsBidAdapter.js:94]: You seem to have forgotten about gpp, usp is deprecated as of Jan 31

**Review by patmmccann — CHANGES_REQUESTED** [review]
Variety of changes needed

**Comment by prBigBrother** [comment]
> Variety of changes needed

Hi @patmmccann.

Thanks for your review!

Right now, the adapter is just like the current one (BrightcomSSP), and we'd rather not make changes this time. How important are the changes you're suggesting?

**Comment by muuki88** [comment]
related docs change https://github.com/prebid/prebid.github.io/pull/5071

File: modules/omsBidAdapter.js (hunk 5 above)
Comment (prBigBrother) [reply modules/omsBidAdapter.js:178]: @patmmccann Can you provide please example how to put it behind the configuration?

File: modules/omsBidAdapter.js (hunk 1 above)
Comment (prBigBrother) [reply modules/omsBidAdapter.js:101]: Fixed, now we getting it from response

File: modules/omsBidAdapter.js (hunk 2 above)
Comment (prBigBrother) [reply modules/omsBidAdapter.js]: we okay to keep it hardcoded

File: modules/omsBidAdapter.js (hunk 3 above)
Comment (prBigBrother) [reply modules/omsBidAdapter.js]: Moved to the library

File: modules/omsBidAdapter.js (hunk 5 above)
Comment (prBigBrother) [reply modules/omsBidAdapter.js:180]: @patmmccann just a friendly reminder. Could you please provide an example as requested?

File: modules/omsBidAdapter.js (hunk 1 above)
Comment (patmmccann) [reply modules/omsBidAdapter.js:101]: you seem to be falling back to the request, please don't call getConfig

File: modules/omsBidAdapter.js (hunk 5 above)
Comment (patmmccann) [reply modules/omsBidAdapter.js:180]: just check the params for something like disableTimeoutReporting and if it is there return true

**Comment by patmmccann** [comment]
>Right now, the adapter is just like the current one (BrightcomSSP), and we'd rather not make changes this time. How important are the changes you're suggesting?

You can't commit new code without review; I don't plan to merge this without some of the feedback addressed

File: modules/omsBidAdapter.js (hunk 1 above)
Comment (prBigBrother) [reply modules/omsBidAdapter.js:101]: Fixed

File: modules/omsBidAdapter.js (hunk 4 above)
Comment (prBigBrother) [reply modules/omsBidAdapter.js]: The `device.sua` object doesn't effectively determine the device type according to our requirements. Here's an example:

For identifying mobile devices, I utilized the user agent associated with the device:

`Mozilla/5.0 (Linux; Android 10; SM-G960F) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.5845.115 Mobile Safari/537.36` 

The resulting sua object would appear as follows:

image

This logic is similarly applied for Apple TV:

`AppleCoreMedia/1.0.0.21K646 (Apple TV; U; CPU OS 17_3 like Mac OS X; pt_pt)` 

image

I understand that the entire logic is based on headers. The only update I propose is to check the `device.sua.mobile ` field in the same line where I check the user agent, as shown below:

``` javascript
 if (sua?.mobile || (/(ios|ipod|ipad|iphone|android)/i).test(ua)) {
    return 1
  }
  ```

File: modules/omsBidAdapter.js (hunk 5 above)
Comment (prBigBrother) [reply modules/omsBidAdapter.js]: We decided to remove this event tracking

File: modules/omsBidAdapter.js [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+        publisher: {
+          id: publisherId
+        }
+      },
+      device: {
+        devicetype: _getDeviceType(),
+        w: screen.width,
+        h: screen.height
+      },
+      tmax: bidderRequest?.timeout
```
[the remark this answers is not in the document]
Comment (prBigBrother) [reply modules/omsBidAdapter.js:87]: Thanks for notice, site.content and user.data was added to bid request

File: modules/omsBidAdapter.js (hunk 6 above)
Comment (prBigBrother) [reply modules/omsBidAdapter.js:74]: we okay with the random identifier string

File: modules/omsBidAdapter.js (hunk 7 above)
Comment (prBigBrother) [reply modules/omsBidAdapter.js:63]: It's okay to keep it like it is

File: modules/omsBidAdapter.js (hunk 8 above)
Comment (prBigBrother) [reply modules/omsBidAdapter.js]: Thanks for notice, GPP handling was added, usp was removed

**Comment by prBigBrother** [comment]
Hey @patmmccann,

I just wanted to follow up on the latest changes. Have you had a chance to review them? Are there any further adjustments needed?

Thanks,
Alex

**Comment by gilditk** [comment]
Hi @patmmccann 
I hope you are well

Will it be possible to comment on our status here?
As you know there are some important updates in this version of ours.

Thanks so much, Gil D
OMS, Product Manager

**Comment by patmmccann** [comment]
taking a look now

**Review by patmmccann — APPROVED** [review]

**Pull request merged**